File: data/data.code.typescript.cc.client/AckSceneEnter.ts
import Packet from '@mi/mod/Packet'
import MsgScenePlayer from './MsgScenePlayer'


export default class AckSceneEnter
{
	private _player: MsgScenePlayer;


	constructor(packet?: Packet) {
		if (packet) {
			this._player = new MsgScenePlayer(packet);
		}
	}


	public get player(): MsgScenePlayer { return this._player; }
	public set player(value: MsgScenePlayer) { this._player = value; }
}

File: data/data.code.lua.server/AckSceneEnter.lua
module("AckSceneEnter", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.player = nil
end


function encode(self)
	local pack = packet.new()
	pack:writeBytes(self.player:getBytes())
	return pack:encode(Msg.P_ACK_SCENE_ENTER)
end
